tests/test_brighterFatterKernel.py

```diff
@@ -0,0 +1,151 @@
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the LSST License Statement and
+# the GNU General Public License along with this program.  If not,
+# see <https://www.lsstcorp.org/LegalNotices/>.
+#
+"""Test cases for lsst.cp.pipe.BrighterFatterKernelSolveTask.
+"""
+
+import unittest
+import numpy as np
+
+import lsst.utils
+import lsst.utils.tests
+
+import lsst.ip.isr as ipIsr
+import lsst.cp.pipe as cpPipe
+import lsst.afw.cameraGeom as cameraGeom
+
+
+class BfkSolveTaskTestCase(lsst.utils.tests.TestCase):
+    """A test case for the brighter fatter kernel solver.
+    """
+
+    def setUp(self):
+        """Set up a plausible PTC dataset, with 1% of the expected variance
+        shifted into covariance terms.
+        """
+        self.defaultConfig = cpPipe.BrighterFatterKernelSolveConfig()
+        self.ptc = ipIsr.PhotonTransferCurveDataset(ampNames=['amp 1'], ptcFitType='FULLCOVARIANCE',
+                                                    covMatrixSide=3)
+        self.ptc.expIdMask['amp 1'] = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9]
+        self.ptc.finalMeans['amp 1'] = [1000, 2000, 3000, 4000, 5000, 6000, 7000, 8000, 9000, 10000]
+        self.ptc.rawMeans['amp 1'] = self.ptc.finalMeans['amp 1']
+        self.ptc.finalVars['amp 1'] = 0.99 * np.array(self.ptc.finalMeans['amp 1'], dtype=float)
+        self.ptc.covariances['amp 1'] = []
+        for mean, variance in zip(self.ptc.finalMeans['amp 1'], self.ptc.finalVars['amp 1']):
+            residual = mean - variance
+            covariance = [[variance, 0.5 * residual, 0.1 * residual],
+                          [0.2 * residual, 0.1 * residual, 0.05 * residual],
+                          [0.025 * residual, 0.015 * residual, 0.01 * residual]]
+            self.ptc.covariances['amp 1'].append(covariance)
+
+        self.ptc.gain['amp 1'] = 1.0
+        self.ptc.noise['amp 1'] = 5.0
+
+        cameraBuilder = cameraGeom.Camera.Builder('fake camera')
+        detectorWrapper = cameraGeom.testUtils.DetectorWrapper(numAmps=1, cameraBuilder=cameraBuilder)
+        self.detector = detectorWrapper.detector
+        self.camera = cameraBuilder.finish()
+
+    def test_averaged(self):
+
+        """Test "averaged" brighter-fatter kernel.
+        """
+        task = cpPipe.BrighterFatterKernelSolveTask()
+
+        results = task.run(self.ptc, ['this is a dummy exposure'], self.camera, {'detector': 1})
+        expectation = np.array([[4.88348887e-08, 1.01136877e-07, 1.51784114e-07,
```

Comment:
How is the expectation obtained? (here and in the other test) Do you have independendent code?

Reply:
This is an expectation generated by running the code on the fake inputs.  It will almost certainly break if there are any algorithmic changes, but this unit test does exercise all of the code (and was useful in tracking down typos and unexpected problems in branches not used on the imsim or LATISS tests).